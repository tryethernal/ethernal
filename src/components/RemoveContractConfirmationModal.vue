<template>
<v-dialog v-model="dialog" max-width="600">
    <v-card>
        <v-card-title class="headline">Remove contract</v-card-title>

        <v-card-text>
            <v-alert type="error" v-if="errorMessage"> {{ errorMessage }}</v-alert>
            <div>
                Removing this contract will remove metadata (name, ABI, AST, hashed bytecode) associated to this address and prevent you from:
                <ul>
                    <li>Interacting with it through Ethernal.</li>
                    <li>Decoding events, function calls, internal calls (past decoded events & calls will be lost).</li>
                    <li>Decoding storage variables (past decoded variables will be lost).</li>
                </ul>
            </div>
        </v-card-text>

        <v-card-actions>
            <v-spacer></v-spacer>
            <v-btn color="primary" text @click="close()">Close</v-btn>
            <v-btn id="removeContract" color="primary" :loading="loading" @click.stop="removeContract()">Remove</v-btn>
        </v-card-actions>
    </v-card>
</v-dialog>
</template>
<script>
export default {
    name: 'RemoveContractConfirmationModal',
    data: () => ({
        dialog: false,
        resolve: null,
        reject: null,
        loading: false,
        address: null,
        workspace: null,
        errorMessage: null
    }),
    methods: {
        open: function(options) {
            this.dialog = true;
            this.address = options.address;
            this.workspace = options.workspace;

            return new Promise((resolve, reject) => {
                this.resolve = resolve;
                this.reject = reject;
            });
        },
        close: function() {
            this.resolve(true);
            this.reset();
        },
        removeContract: function() {
            this.loading = true;
            this.server.removeContract(this.address)
                .then(() => {
                    if (this.$router.currentRoute.path != '/contracts')
                        this.$router.push({ path: '/contracts', query: { removedContract: this.address }});
                    this.close();
                })
                .catch((error) => {
                    this.errorMessage = error.message;
                    this.loading = false;
                });

        },
        reset: function() {
            this.address = null;
            this.workspace = null;
            this.dialog = false;
            this.reject = null;
            this.resolve = null;
            this.loading = false;
            this.errorMessage = null;
        }
    }
}
</script>
