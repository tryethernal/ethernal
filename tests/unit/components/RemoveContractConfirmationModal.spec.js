import MockHelper from '../MockHelper';
  
import RemoveContractConfirmationModal from '@/components/RemoveContractConfirmationModal.vue';

describe('RemoveContractConfirmationModal.vue', () => {
    let helper;

    beforeEach(() => {
        helper = new MockHelper();
    });

    it('Should display the modal', async () => {
        const wrapper = helper.mountFn(RemoveContractConfirmationModal);
        await wrapper.setData({ dialog: true, workspace: 'hardhat', address: '0x123' });
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should display an error message if something is wrong', async () => {
        jest.spyOn(helper.mocks.server, 'removeContract').mockRejectedValue({ message: 'There is an error' });

        const wrapper = helper.mountFn(RemoveContractConfirmationModal);
        await wrapper.setData({ dialog: true, workspace: 'hardhat', address: '0x123' });

        await wrapper.find('#removeContract').trigger('click');
        await wrapper.vm.$nextTick();
        
        expect(wrapper.html()).toMatchSnapshot();
    });

    it('Should call the removal function', async () => {
        jest.spyOn(helper.mocks.server, 'removeContract').mockResolvedValue();
        const wrapper = helper.mountFn(RemoveContractConfirmationModal);
        await wrapper.setData({ dialog: true, workspace: 'hardhat', address: '0x123' });

        const removeContractSpy = jest.spyOn(helper.mocks.server, 'removeContract');

        await wrapper.find('#removeContract').trigger('click');
        await wrapper.vm.$nextTick();

        expect(removeContractSpy).toHaveBeenCalled();
    });
});
