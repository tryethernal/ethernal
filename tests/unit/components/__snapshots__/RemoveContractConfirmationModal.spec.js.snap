// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveContractConfirmationModal.vue Should display an error message if something is wrong 1`] = `
<div class="v-dialog__container">
  <div role="dialog" aria-modal="true" class="v-dialog__content v-dialog__content--active" style="z-index: 202;">
    <transition-stub name="dialog-transition" appear="true">
      <div tabindex="0" class="v-dialog v-dialog--active" style="transform-origin: center center; max-width: 600px;">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title headline">Remove contract</div>
          <div class="v-card__text">
            <div role="alert" class="v-alert v-sheet theme--dark error">
              <div class="v-alert__wrapper"><i aria-hidden="true" class="v-icon notranslate v-alert__icon mdi mdi-alert theme--dark"></i>
                <div class="v-alert__content"> There is an error</div>
              </div>
            </div>
            <div>
              Removing this contract will remove metadata (name, ABI, AST, hashed bytecode) associated to this address and prevent you from:
              <ul>
                <li>Interacting with it through Ethernal.</li>
                <li>Decoding events, function calls, internal calls (past decoded events &amp; calls will be lost).</li>
                <li>Decoding storage variables (past decoded variables will be lost).</li>
              </ul>
            </div>
          </div>
          <div class="v-card__actions">
            <div class="spacer"></div> <button type="button" class="v-btn v-btn--text theme--light v-size--default primary--text"><span class="v-btn__content">Close</span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary" id="removeContract"><span class="v-btn__content">Remove</span></button>
          </div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>
`;

exports[`RemoveContractConfirmationModal.vue Should display the modal 1`] = `
<div class="v-dialog__container">
  <div role="dialog" aria-modal="true" class="v-dialog__content v-dialog__content--active" style="z-index: 202;">
    <transition-stub name="dialog-transition" appear="true">
      <div tabindex="0" class="v-dialog v-dialog--active" style="transform-origin: center center; max-width: 600px;">
        <div class="v-card v-sheet theme--light">
          <div class="v-card__title headline">Remove contract</div>
          <div class="v-card__text">
            <!---->
            <div>
              Removing this contract will remove metadata (name, ABI, AST, hashed bytecode) associated to this address and prevent you from:
              <ul>
                <li>Interacting with it through Ethernal.</li>
                <li>Decoding events, function calls, internal calls (past decoded events &amp; calls will be lost).</li>
                <li>Decoding storage variables (past decoded variables will be lost).</li>
              </ul>
            </div>
          </div>
          <div class="v-card__actions">
            <div class="spacer"></div> <button type="button" class="v-btn v-btn--text theme--light v-size--default primary--text"><span class="v-btn__content">Close</span></button> <button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default primary" id="removeContract"><span class="v-btn__content">Remove</span></button>
          </div>
        </div>
      </div>
    </transition-stub>
  </div>
</div>
`;
